src/map/atcommand.c

```diff
@@ -7746,10 +7748,11 @@ ACMD(leave) {
 }
 
 ACMD(accept) {
-	if (!duel->checktime(sd)) {
+	int64 diff = duel->checktime(sd);
+	if (diff > 0) {
 		char output[CHAT_SIZE_MAX];
-		// "Duel: You can take part in duel only one time per %d minutes."
-		sprintf(output, msg_fd(fd,356), battle_config.duel_time_interval);
+		// "Duel: You can take part in duel again after %d seconds."
+		sprintf(output, msg_fd(fd,356), diff);
```

Comment:
Good that you pointed that out, for a more architectural compliant data type you can use `time_t`. As for the duel setting limit, it's just better to leave it flexible.